client-index/src/auto_synchronizer.rs

```diff
@@ -92,45 +82,55 @@ impl AutoSynchronizer {
+            let mut channel_sink = channel_tx.clone().wait();
 
-        let runner = ClientBuilder::new(&self.websocket_url)
-            .expect("client-builder new")
-            .add_protocol("rust-websocket")
-            .async_connect_insecure()
-            .and_then(|(duplex, _)| {
-                channel_sink
-                    .send(OwnedMessage::Text(CMD_SUBSCRIBE.to_string()))
-                    .expect("send to channel sink");
-                let (sink, stream) = duplex.split();
+            let runner = ClientBuilder::new(&self.websocket_url)
+                .expect("client-builder new")
+                .add_protocol("rust-websocket")
+                .async_connect_insecure()
+                .and_then(|(duplex, _)| {
+                    channel_sink
+                        .send(OwnedMessage::Text(CMD_SUBSCRIBE.to_string()))
+                        .expect("send to channel sink");
+                    let (sink, stream) = duplex.split();
 
-                stream
-                    .filter_map(|message| match message {
-                        OwnedMessage::Text(a) => {
-                            if let Some(core) = self.core.as_ref() {
-                                core.send(OwnedMessage::Text(a.clone())).expect("core send");
-                            }
+                    stream
+                        .filter_map(|message| match message {
+                            OwnedMessage::Text(a) => {
+                                if let Some(core) = self.core.as_ref() {
+                                    core.send(OwnedMessage::Text(a.clone())).expect("core send");
+                                }
 
-                            None
-                        }
-                        OwnedMessage::Binary(_a) => None,
-                        OwnedMessage::Close(e) => Some(OwnedMessage::Close(e)),
-                        OwnedMessage::Ping(d) => Some(OwnedMessage::Pong(d)),
-                        _ => None,
-                    })
-                    .select(channel_rx.map_err(|_| WebSocketError::NoDataAvailable))
-                    .forward(sink)
-            });
-        let _ = runtime.block_on(runner).expect("tokio block_on");
-        Ok(())
+                                None
+                            }
+                            OwnedMessage::Binary(_a) => None,
+                            OwnedMessage::Close(e) => Some(OwnedMessage::Close(e)),
+                            OwnedMessage::Ping(d) => Some(OwnedMessage::Pong(d)),
+                            _ => None,
+                        })
+                        .select(channel_rx.map_err(|_| WebSocketError::NoDataAvailable))
+                        .forward(sink)
+                });
+            match runtime.block_on(runner) {
+                Ok(_a) => {}
+                Err(b) => {
+                    println!("connection fail {}", b);
+                    std::thread::sleep(std::time::Duration::from_millis(1000));
```

Comment:
this will spam the error log -- the retry time could be extended on each retry attempt and/or there could be a maximum number of retries (after which it'll crash the whole client-rpc process)